src/services/iiif-parser.js

```diff
@@ -51,83 +52,193 @@ export function getChildCanvases({ rangeId, manifest }) {
+  let annotations = [];
+  // When annotations are at canvas level
+  const annotationPage = parseManifest(manifest)
+    .getSequences()[0]
+    .getCanvases()[canvasIndex];
+
+  if (annotationPage) {
+    annotations = parseAnnotations(annotationPage.__jsonld.items, 'painting');
+  }
+  return annotations;
+}
+
 /**
  * Get sources and media type for a given canvas
  * If there are no items, an error is returned (user facing error)
  * @param {Object} obj
  * @param {Object} obj.manifest IIIF Manifest
  * @param {Number} obj.canvasIndex Index of the current canvas in manifest
- * @returns {Array.<Object>} array of file choice objects
+ * @returns {Array.<Object>} array of objects
  */
-export function getMediaInfo({ manifest, canvasIndex }) {
-  let choiceItems,
-    sources = [],
-    tracks = [];
-  let isSelected = false;
+export function getMediaInfo({ manifest, canvasIndex, srcIndex }) {
+  let canvas = [];
+
+  // return empty object when canvasIndex is undefined
+  if (canvasIndex === undefined || canvasIndex < 0) {
+    return { error: 'Error fetching content' };
+  }
 
+  // Get the canvas with the given canvasIndex
   try {
-    choiceItems = parseManifest(manifest)
+    canvas = parseManifest(manifest)
       .getSequences()[0]
-      .getCanvases()
-      [canvasIndex].getContent()[0]
-      .getBody();
+      .getCanvasByIndex(canvasIndex);
   } catch (e) {
-    console.log('error fetching content', e);
-    return { error: 'Error fetching content' };
+    console.log('Error fetching resources: ', e);
+    return { error: 'Error fetching resources' };
   }
 
-  if (choiceItems.length === 0) {
+  // Canvas properties
+  const canvasProps = {
+    duration: canvas.getDuration(),
+    height: canvas.getHeight(),
+    width: canvas.getWidth(),
+  };
+
+  const annotations = getAnnotations({
+    manifest,
+    canvasIndex,
+  });
+
+  const mediaInfo = getResourceItems(
+    annotations,
+    srcIndex,
+    canvasProps.duration
+  );
+
+  mediaInfo.canvas = canvasProps;
+
+  if (mediaInfo.error) {
+    return { ...mediaInfo };
+  } else {
+    // Get media type
+    let allTypes = mediaInfo.sources.map((q) => q.kind);
+    const mediaType = setMediaType(allTypes);
     return {
-      error: 'No media sources found',
+      ...mediaInfo,
+      error: null,
+      mediaType,
     };
+  }
+}
+
+function getResourceItems(annotations, srcIndex, duration) {
+  let sources = [],
+    tracks = [],
+    canvasTargets = [],
+    isMultiSource = false;
+
+  if (annotations.length === 0) {
+    return { error: 'No resources found in Manifest' };
+  } else if (annotations.length > 1) {
+    isMultiSource = true;
+    annotations.map((a, index) => {
+      const { source, track } = getResourceInfo(a.getBody()[0]);
+      const target = parseCanvasTarget(a, duration, index);
+      canvasTargets.push(target);
+      source.length > 0 && sources.push(source[0]);
+      track.length > 0 && tracks.push(track[0]);
```

Comment:
Yes, it would mess things up, even though it probably would not break the player.

I wasn't sure how this scenario would be represented in a manifest. It would be great if we can find manifests that uses multiple sources with text annotations, so that it's easier to understand and build the logic.

It will need some time to figure this out. 

Maybe I can put this as a comment in the code and mark it as future work?

Reply:
I think that's fine since it is an edge case right now.